test/saithrift/tests/saitunnel.py

```diff
@@ -0,0 +1,321 @@
+        
+        #Packet to be send 
+        pkt = simple_tcp_packet(eth_dst=router_mac,
+                                eth_src='00:00:00:00:00:01',
+                                ip_src='20.20.1.2',
+                                ip_dst='1.1.1.1',
+                                ip_id=1,
+                                ip_ttl=64)
+        #expected packet inner IP header
+        inner_hdr = simple_tcp_packet(eth_dst=router_mac,
+                                eth_src='00:00:00:00:00:01',
+                                ip_src='20.20.1.2',
+                                ip_dst='1.1.1.1',
+                                ip_id=1,
+                                ip_ttl=63)
+        exp_pkt = simple_ipv4ip_packet(eth_dst='00:05:06:00:00:00',
+                                eth_src=router_mac,
+                                ip_dst='10.10.1.1',
+                                ip_src='2.2.2.2',
+                                ip_id=0,#mask the indentifier during check because it differs evry time.better chech with wireshark
+                                ip_ttl=64,
+                                inner_frame=inner_hdr['IP']
+                                )
+        #masking packet 
+        m=Mask(exp_pkt)
+        m.set_do_not_care_scapy(ptf.packet.IP , 'id')
+        
+        try:
+            # in tuple: 0 is device number, 1 is port number
+            # this tuple uniquely identifies a port
+            send_packet(self, 1, pkt)
+            #verify_packets(self,exp_pkt,ports=[0])
+            verify_packets(self,str(m),[0])
+
+        
+        finally:
+            sai_thrift_remove_route(self.client,vr_id,addr_family,tunnel_ip_addr,tunnel_ip_mask,initiator_id) 
+            self.client.sai_thrift_remove_next_hop(initiator_id)
+
+            sai_thrift_remove_route(self.client,vr_id,addr_family,nhop_ip_addr_encap,nhop_ip_mask_encap,egress_nhop) 
+            self.client.sai_thrift_remove_next_hop(egress_nhop)
+
+            sai_thrift_remove_route(self.client,vr_id,addr_family,nhop_ip_addr_ingress,nhop_ip_mask_ingress,ingress_nhop) 
+            self.client.sai_thrift_remove_next_hop(ingress_nhop)
+
+            sai_thrift_remove_neighbor(self.client,addr_family,rif_id=ingress_if_id,ip_addr=ingress_ip_addr,dmac=neighbor_mac_ingress)
+            
+            sai_thrift_remove_neighbor(self.client,addr_family,rif_id=encap_if_id,ip_addr=encap_ip_addr,dmac=neighbor_mac_encap)
+            self.client.sai_thrift_remove_router_interface(ingress_if_id)
+
+            self.client.sai_thrift_remove_router_interface(encap_if_id)
+
+            self.client.sai_thrift_remove_tunnel(tunnel_id)
+
+
+@group("tunnel")
+@disabled
+class decapsulation(sai_base_test.ThriftInterfaceDataPlane):
+    '''
+    performs the decapsulation 
+    Tunnel table source is 10.10.1.1
+    Tunnel table Dest is 2.2.2.2
+    
+    We will send encapsulated packet from port1 and expects a decapsulated packet on port2

```

Comment:
can you describe the test in more detail? For example, what the inner and outer packet dst ip, what is the underlay routing table, what is the overlay tunnel table? What is the ECN mode, TTL mode?